hips/hip-0019.md

```diff
@@ -0,0 +1,209 @@
+---
+hip: 0019
+title: "Enable Running Hooks in Parallel"
+authors: [ "Michael Morris <michael.morris@est.tech>" ]
+created: "2024-06-06"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+Helm hooks are currently executed serially. In many cases there will be no reason why they cannot be run in parallel. Running hooks in parallel can significantly improve install/upgrade times
+
+## Motivation
+
+Decrease install/upgrade times by executing hooks in parallel
+
+## Specification
+
+Only hooks of the same weight may be executed in parallel. The chart designer can specify if hooks will be run in parallel. I am proposing two options, one more fine grained that the other. 
+
+Option 1.
+Add a new parameter to the chart metadata of type boolean with name 'runHooksInParallel'.
+When set to 'false' (default), the behavior will be as today (hooks run in sequence).
+When set to 'true', hooks of the same weight for the chart shall be run in parallel.
+In the case of subcharts, hooks of the same weight from different sub charts with 'runHooksInParallel' set to 'true' will be run in parallel. Hooks from subcharts with 'runHooksInParallel' to 'false' shall be run sequentially after hooks of the same weight that are permitted to run in parallel.
+
+Option 2.
+Add a new parameter to the chart metadata with name 'runHooksInParallel' that accepts three values - 'false', 'otherChartsOnly' and 'true'.
+Behavior for 'true' and 'false' is as per option 1. above.
+The value 'otherChartsOnly' is only relevant in the case of subcharts and shall indicate that the hooks belonging to the particular subchart have to be run sequentially but, it is ok for this to happen in parallel with the execution of hooks from other subcharts
```

Comment:
I'm not really seeing a use-case for "otherChartsOnly".

Why wouldn't a subchart just adjust its weights to be sequential, it it wanted its hooks to run sequentially?. Wouldn't that achieve the same as "otherChartsOnly"? Perhaps there is a scenario I'm not thinking of?

Reply:
Really just to make migration easier (as outlined in above reply), but yes, I agree, should use hook weights for this scenario instead